src/sagemaker/jumpstart/curated_hub/jumpstart_curated_public_hub.py

```diff
@@ -0,0 +1,362 @@
+        # TODO enable: self.studio_metadata_map[model_specs.model_id]["desc"]
+        hub_content_description = "This is a curated model based "
+        +f"off the public JumpStart model {hub_content_display_name}"
+        hub_content_markdown = self._dst_s3_filesystem.get_markdown_s3_reference(
+            model_specs
+        ).get_uri()
+
+        hub_content_document = self._document_creator.make_hub_content_document(
+            model_specs=model_specs
+        )
+
+        self._sm_client.import_hub_content(
+            HubName=self.curated_hub_name,
+            HubContentName=model_specs.model_id,
+            HubContentVersion=model_specs.version,
+            HubContentType="Model",
+            DocumentSchemaVersion="1.0.0",
+            HubContentDisplayName=hub_content_display_name,
+            HubContentDescription=hub_content_description,
+            HubContentMarkdown=hub_content_markdown,
+            HubContentDocument=hub_content_document,
+        )
+
+    def delete_models(self, model_ids: List[PublicModelId]):
+        """Deletes all versions of each model"""
+        model_specs = self._get_model_specs(model_ids)
+        for model_spec in model_specs:
+            self._delete_model_from_curated_hub(model_spec)
+
+    def _delete_model_from_curated_hub(
+        self, model_specs: JumpStartModelSpecs, delete_dependencies: bool = True
+    ):
+        if delete_dependencies:
+            self._delete_model_dependencies_no_content_noop(model_specs)
+        self._hub_client.delete_all_versions_of_model(model_specs)
+
+    def _delete_model_dependencies_no_content_noop(self, model_specs: JumpStartModelSpecs):
+        try:
+            hub_content = self._sm_client.describe_hub_content(
+                HubName=self.curated_hub_name,
+                HubContentName=model_specs.model_id,
+                HubContentVersion=model_specs.version,
+                HubContentType="Model",
+            )
+        except ClientError:
+            return
+
+        dependencies = self._get_hub_content_dependencies_from_model_document(
+            hub_content["HubContentDocument"]
+        )
+        dependency_s3_keys = []
+        for dependency in dependencies:
+            dependency_s3_keys.extend(
+                self._format_dependency_dst_uris_for_delete_objects(dependency)
+            )
+        print(f"Deleting HubContent dependencies for {model_specs.model_id}: {dependency_s3_keys}")
+        delete_response = self._s3_client.delete_objects(
+            Bucket=self.curated_hub_s3_bucket_name,
+            Delete={"Objects": dependency_s3_keys, "Quiet": True},
+        )
+
+        if "Errors" in delete_response:
+            raise Exception(
+                f"Failed to delete all dependencies of model {model_specs.model_id}. : {delete_response['Errors']}"
+            )
+
+    def _get_hub_content_dependencies_from_model_document(
```

Comment:
we should have a data class for the hub content document, ideally, to make it easier for other parts of the code to use it